composeApp/src/commonMain/kotlin/presentation/screens/details/ShareSheet.kt

```diff
@@ -0,0 +1,184 @@
+import hackernewskmp.composeapp.generated.resources.ic_square_share_line_linear
+import hackernewskmp.composeapp.generated.resources.ic_square_top_down_linear
+import hackernewskmp.composeapp.generated.resources.open_with_the_default_browser
+import hackernewskmp.composeapp.generated.resources.open_with_x
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.DrawableResource
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+import presentation.screens.main.previewItems
+import ui.AppPreview
+import ui.googleSansCodeFontFamily
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalComposeUiApi::class)
+@Composable
+fun DetailsShareSheet(
+    isVisible: Boolean,
+    item: Item,
+    onOpenInBrowser: () -> Unit,
+    onShareLink: () -> Unit,
+    onShareComments: () -> Unit,
+    onVisibility: (Boolean) -> Unit,
+) {
+    val scope = rememberCoroutineScope()
+    val bottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+
+    LaunchedEffect(isVisible) {
+        scope.launch {
+            if (isVisible && !bottomSheetState.isVisible) {
+                bottomSheetState.expand()
+            }
+            else if (!isVisible && bottomSheetState.isVisible) {
+                bottomSheetState.hide()
+            }
+        }
+    }
+
+    LaunchedEffect(bottomSheetState.isVisible) {
+        onVisibility(bottomSheetState.isVisible)
+    }
+
+    if (isVisible) {
+        ModalBottomSheet(
+            sheetState = bottomSheetState,
+            containerColor = MaterialTheme.colorScheme.surfaceContainer,
+            onDismissRequest = {},
+            content = {
+                SheetContent(
+                    item = item,
+                    onOpenInBrowser = onOpenInBrowser,
+                    onShareLink = onShareLink,
+                    onShareComments = onShareComments,
+                )
+            },
+        )
+
+        // Handle Back Gesture / Back Press to collapse sheet
+        BackHandler {
+            scope.launch {
+                bottomSheetState.hide()
+            }
+        }
+    }
+}
+
+@Composable
+private fun SheetContent(
+    item: Item,
+    onOpenInBrowser: () -> Unit,
+    onShareLink: () -> Unit,
+    onShareComments: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 24.dp, end = 24.dp, bottom = 24.dp),
+    ) {
+        item.getUrl()?.let { urlString ->
+            val defaultBrowserName = remember(urlString) { getPlatform().getDefaultBrowserName(urlString) }
+            val buttonText = when {
+                defaultBrowserName == null -> stringResource(Res.string.open_with_the_default_browser)
+                else -> stringResource(Res.string.open_with_x)
+                    .replace("{browser-name}", defaultBrowserName)
```

Comment:
Instead of using `.replace("{browser-name}", defaultBrowserName)`, it's better to use string formatting with placeholders in the resource itself.  

For example, define the string in `strings.xml` like:

`<string name="open_with_x">Open with %1$s</string>`

Then use it in code:

`stringResource(Res.string.open_with_x, defaultBrowserName)`